recipes/validate.rb

```diff
@@ -78,6 +78,11 @@
   node.run_state['issues_detected'] = true
 end
 
+if first_etcd.nil?
```

Comment:
You forgot to set **`first_etcd = server_info.first_etcd`**
In addition, that check is valid only when deploying in cluster architecture. 
Therefore you should add : 
**only_if { node['cookbook-openshift3']['openshift_HA'] }**